Functions/Solution.cs

```diff
@@ -15,74 +16,267 @@ internal class Solution
+            {
+                Console.WriteLine("Not a legal move.");
+            }
         }
 
         public static double[] Assignment4(bool asc, params double[] nums)
         {
-            /*
-             * TODO: Write code here
-             */
+            int n = nums.Length;
+            double[] arr = new double[n];
+            
+            nums.CopyTo(arr, 0);
+
+            double tmp;
+            for(int i=0; i<n; i++)
+            {
+                for(int j=i+1; j<n; j++)
+                {
+                    if(arr[j] < arr[i])
+                    {
+                        tmp = arr[i];
+                        arr[i] = arr[j];
+                        arr[j] = tmp;
+                    }
+                }
+            }
+
+            // change sort direction if needed
+            if (asc == false)
+            {
+                double[] final = new double[n];
+                for(int i=0; i<n; i++)
+                    final[i] = arr[n-1-i];
+                return final;
+            }
+            else
+                return arr;
         }
 
         public static void Assignment5(string str, char[] letters)
         {
-            /*
-             * TODO: Write code here
-             */
+            const int letterOffset = 10;
+            var newStr = new StringBuilder();
+            foreach (char c in str)
+            {
+                bool charInLetters = false;
+                Console.WriteLine();
+                foreach (char letter in letters)
+                {
+                    if (c == letter)
+                    {
+                        charInLetters = true;
+                        if (c + letterOffset > 'z')
+                            {
+                                newStr.Append((char)('a' + 'z' - c));
+                                break;
+                            }
+                        else
+                            newStr.Append((char)(c+letterOffset));
+                    }
+                }
+                if (!charInLetters)
+                    newStr.Append(c);
+            }
+            string changedStr = newStr.ToString();
+            Console.WriteLine(changedStr);
         }
 
         public static bool Assignment6(string str)
-        {
-            /*
-             * TODO: Write code here
-             */
+        {   
+            Regex rgx = new Regex("[^a-zA-Z]");
+            str = rgx.Replace(str, "");
+            string sentence = str.ToLower();
+            int n = sentence.Length - 1;
+            bool isPalindrome = true;
+            int i = 0;
+            while(i < n-i)
+            {
+                if (sentence[i] != sentence[n-i])
+                {
+                    isPalindrome = false;
+                    break;
+                }
+                i++;
+            }
+            return isPalindrome;
         }
 
         public static int Assignment7(int[] nums)
```

Comment:
This is a nice approach with `O(n)` complexity for time and space. I would like to challenge you to think about an `O(n)` time complexity algorithm with `O(1)` space.